From 8f87b4288f3c9a7c7a3fa425b6070cccb0a81688 Mon Sep 17 00:00:00 2001
From: Moudy Ho <moudy.ho@mediatek.com>
Date: Wed, 17 Dec 2025 08:58:58 +0800
Subject: [PATCH] GENIO: arm64: dts: mt8189: update GPU clock configuration

Previously, only the "subsys_bg3d" clock was defined for the GPU node.
The new configuration adds multiple clocks to better reflect the GPU's
clock hierarchy and dependencies, including:

- clk_mux: GPU clock multiplexer
- clk_pll_src: PLL source for GPU
- clk_main_parent: Main parent clock for GPU
- clk_sub_parent: Sub-parent clock for GPU
- subsys_bg3d: Subsystem clock for GPU

These changes ensure the GPU node is properly configured with all necessary
clocks, avoiding unexpected DVFS errors by ensuring all relevant clock
sources are correctly defined.

Change-Id: I7d2486e5ea42b5e4d5e7d31095110dffe0351051
Signed-off-by: Moudy Ho <moudy.ho@mediatek.com>
---
 arch/arm64/boot/dts/mediatek/mt8189.dtsi | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt8189.dtsi b/arch/arm64/boot/dts/mediatek/mt8189.dtsi
index 27e2064d1a0b..26e665f54778 100644
--- a/arch/arm64/boot/dts/mediatek/mt8189.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8189.dtsi
@@ -1132,8 +1132,16 @@ gpu: gpu@13000000 {
 				     <GIC_SPI 269 IRQ_TYPE_LEVEL_HIGH 0>;
 			interrupt-names = "job", "mmu", "gpu";
 
-			clocks = <&mfg_clk CLK_MFG_BG3D>;
-			clock-names = "subsys_bg3d";
+			clocks = <&topckgen_clk CLK_TOP_MFG_SEL_MFGPLL>,
+				 <&apmixedsys_clk CLK_APMIXED_MFGPLL>,
+				 <&topckgen_clk CLK_TOP_MFGPLL>,
+				 <&topckgen_clk CLK_TOP_MFG_REF_SEL>,
+				 <&mfg_clk CLK_MFG_BG3D>;
+			clock-names = "clk_mux",
+				      "clk_pll_src",
+				      "clk_main_parent",
+				      "clk_sub_parent",
+				      "subsys_bg3d";
 
 			operating-points-v2 = <&gpu_opp_table>;
 			power-domains = <&scpsys MT8189_POWER_DOMAIN_MFG2>,
-- 
2.45.2

